src/main/java/com/o19s/es/ltr/rest/RestSimpleFeatureStore.java

```diff
@@ -0,0 +1,249 @@
+        // Validation happens here when parsing the stored element.
+        if (!elt.name().equals(name)) {
+            throw new IllegalArgumentException("Name mismatch, send request with [" + elt.name() + "] but [" + name + "] used in the URL");
+        }
+        if (request.method() == RestRequest.Method.POST && !elt.updatable()) {
+            throw new IllegalArgumentException("Element of type [" + elt.type() + "] are not updatable.");
+        }
+        FeatureStoreAction.FeatureStoreRequestBuilder builder = FeatureStoreAction.INSTANCE.newRequestBuilder(client);
+        if (request.method() == RestRequest.Method.PUT) {
+            builder.request().setAction(FeatureStoreAction.FeatureStoreRequest.Action.CREATE);
+        } else {
+            builder.request().setAction(FeatureStoreAction.FeatureStoreRequest.Action.UPDATE);
+        }
+        builder.request().setStorableElement(elt);
+        builder.request().setRouting(routing);
+        builder.request().setStore(indexName);
+        return (channel) -> builder.execute(new RestStatusToXContentListener<FeatureStoreAction.FeatureStoreResponse>(channel,
+                (r) -> r.getResponse().getLocation(routing)));
+    }
+
+
+    RestChannelConsumer delete(NodeClient client, String type, String indexName, RestRequest request) {
+        assert SUPPORTED_TYPES.contains(type);
+        String name = request.param("name");
+        String id = generateId(type, name);
+        String routing = request.param("routing");
+        return (channel) ->  client.prepareDelete(indexName, ES_TYPE, id).setRouting(routing)
+                .execute(new RestStatusToXContentListener<>(channel, (r) -> r.getLocation(routing)));
+    }
+
+    RestChannelConsumer get(NodeClient client, String type, String indexName, RestRequest request) {
+        assert SUPPORTED_TYPES.contains(type);
+        String name = request.param("name");
+        String routing = request.param("routing");
+        String id = generateId(type, name);
+        return (channel) -> client.prepareGet(indexName, ES_TYPE, id)
+                .setRouting(routing)
+                .execute(new RestToXContentListener<GetResponse>(channel) {
+                    @Override
+                    protected RestStatus getStatus(final GetResponse response) {
+                        return response.isExists() ? OK : NOT_FOUND;
+                    }
+                });
+    }
+
+    RestChannelConsumer search(NodeClient client, String type, String indexName, RestRequest request) {
+        String prefix = request.param("prefix");
+        int from = request.paramAsInt("from", 0);
+        int size = request.paramAsInt("size", 20);
+        BoolQueryBuilder qb = QueryBuilders.boolQuery().filter(QueryBuilders.termQuery("type", type));
+        if (prefix != null && !prefix.isEmpty()) {
+            qb.must(QueryBuilders.matchQuery("name.prefix", prefix));
```

Comment:
A static import of QueryBuilders.matchQuery would make this read more fluently (very minor, feel free to ignore).